src/arrayMethodJoin.js

```diff
@@ -4,8 +4,29 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    this.forEach((element, index) => {
+      let elementToAdd = element;
+      let separatorToAdd = separator;
+
+      if (separator === null) {
+        separatorToAdd = 'null';
+      }
+
+      if (elementToAdd === undefined || elementToAdd === null) {
+        elementToAdd = '';
+      }
+
+      if (index === this.length - 1) {
+        string += elementToAdd;
+      } else {
+        string += elementToAdd + separatorToAdd;
+      }
```

Comment:
you add `elementToAdd` in each case, so there is no need to add it in `if` statement
```suggestion
      string += elementToAdd;
      
      if (index !== this.length - 1) {
        string += separatorToAdd;
      }
```